src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/bolts/HaFlowPathSwapHubBolt.java

```diff
@@ -0,0 +1,193 @@
+import org.apache.storm.tuple.Values;
+
+public class HaFlowPathSwapHubBolt extends HubBolt implements FlowPathSwapHubCarrier {
+
+    private final HaFlowPathSwapConfig config;
+    private final RuleManagerConfig ruleManagerConfig;
+
+    private transient HaFlowPathSwapService service;
+    private String currentKey;
+
+    private LifecycleEvent deferredShutdownEvent;
+
+    public HaFlowPathSwapHubBolt(
+            @NonNull HaFlowPathSwapHubBolt.HaFlowPathSwapConfig config, @NonNull PersistenceManager persistenceManager,
+            @NonNull RuleManagerConfig ruleManagerConfig) {
+        super(persistenceManager, config);
+
+        this.config = config;
+        this.ruleManagerConfig = ruleManagerConfig;
+
+        enableMeterRegistry("kilda.ha_flow_pathswap", HUB_TO_METRICS_BOLT.name());
+    }
+
+    @Override
+    protected void init() {
+        RuleManager ruleManager = new RuleManagerImpl(ruleManagerConfig);
+        service = new HaFlowPathSwapService(this, persistenceManager, ruleManager, config.getSpeakerCommandRetriesLimit());
+    }
+
+    @Override
+
+    protected boolean deactivate(LifecycleEvent event) {
+        if (service.deactivate()) {
+            return true;
+        }
+        deferredShutdownEvent = event;
+        return false;
+    }
+
+    @Override
+    protected void activate() {
+        service.activate();
+    }
+
+    @Override
+    protected void onRequest(Tuple input) throws PipelineException {
+        currentKey = input.getStringByField(MessageKafkaTranslator.FIELD_ID_KEY);
+        HaFlowPathSwapRequest payload = (HaFlowPathSwapRequest) input.getValueByField(FIELD_ID_PAYLOAD);
+        try {
+            service.handleRequest(currentKey, pullContext(input), payload.getHaFlowId());
+        } catch (DuplicateKeyException e) {
+            log.error("Failed to handle a request with key {}. {}", currentKey, e.getMessage());
+        }
+    }
+
+    @Override
+    protected void onWorkerResponse(Tuple input) {
+        String operationKey = input.getStringByField(MessageKafkaTranslator.FIELD_ID_KEY);
+        currentKey = KeyProvider.getParentKey(operationKey);
+        SpeakerCommandResponse flowResponse = (SpeakerCommandResponse) input.getValueByField(FIELD_ID_PAYLOAD);
+        service.handleAsyncResponse(currentKey, flowResponse);
+    }
+
+    @Override
+    public void onTimeout(@NonNull String key, Tuple tuple) {
+        currentKey = key;
+        service.handleTimeout(key);
+    }
+
+    @Override
+    public void sendInactive() {
+        getOutput().emit(ZkStreams.ZK.toString(), new Values(deferredShutdownEvent, getCommandContext()));
+        deferredShutdownEvent = null;
+    }
+
+    @Override
+    public void sendPeriodicPingNotification(String haFlowId, boolean enabled) {
+        //TODO implement periodic pings https://github.com/telstra/open-kilda/issues/5153
+        log.info("Periodic pings are not implemented for ha-flow swap path operation yet. Skipping for the ha-flow {}",
```

Comment:
In some places you use "HA-Flow", in other "HA-flow, here "ha-flow".

Reply:
fixed